cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -146,6 +143,71 @@ func WriteStaticPodManifests(cfg *kubeadmapi.MasterConfiguration) error {
 	return nil
 }
 
+// getHostPathVolumesForTheControlPlane gets the volumes and volumemounts needed for each control plane component
+func getHostPathVolumesForTheControlPlane(cfg *kubeadmapi.MasterConfiguration) (map[string][]v1.Volume, map[string][]v1.VolumeMount) {
```

Comment:
I'd consider just create a struct that hides the details, this function is a little hard to read.  Someone not intimately familiar with what is going here would probably get confused.